src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import decimal
+import inspect
+import re
+import sys
+
+from functools import reduce
+from pathlib import Path
+from typing import Any, List, Optional
+
+import snowflake.snowpark._internal.proto.ast_pb2 as proto
+
+NoneType = type(None)
+PYTHON_TO_AST_CONST_MAPPINGS = {
+    NoneType: "null_val",
+    bool: "bool_val",
+    int: "int64_val",
+    float: "float64_val",
+    str: "string_val",
+    bytearray: "binary_val",
+    bytes: "binary_val",
+    decimal.Decimal: "big_decimal_val",
+    datetime.date: "date_val",
+    datetime.time: "python_time_val",
+    datetime.datetime: "python_timestamp_val",
+}
+def infer_const_ast(obj: Any, ast: proto.Expr) -> None:
```

Comment:
nit: maybe rename to fill_const_ast? I think I saw you define another fill API somewhere.